src/App.tsx

```diff
@@ -34,62 +47,133 @@ export function getReorderedGoods(
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+export class App extends React.Component<{}, State> {
+  state = {
+    isReversed: false,
+    // eslint-disable-next-line react/no-unused-state
+    sortType: true,
+    sortBy: 'id',
+    isShowm: false,
+    isClickedAlph: true,
+    isClickedLength: true,
+    isClickedReverse: true,
+  };
 
-      <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
-      </ul>
-    </div>
-  );
-};
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+      isShowm: true,
+      isClickedReverse: !state.isClickedReverse,
+
+    }));
+  };
+
+  sortByAlphabet = () => {
+    this.setState({
+      sortBy: 'alphabet',
+      isShowm: true,
+      isClickedAlph: false,
+      isClickedLength: true,
+    });
+  };
+
+  sortByLength = () => {
+    this.setState({
+      sortBy: 'length',
+      isShowm: true,
+      isClickedLength: false,
+      isClickedAlph: true,
+    });
+  };
+
+  reset = () => {
+    this.setState({
+      sortBy: 'id',
+      isReversed: false,
+      isShowm: false,
+      isClickedAlph: true,
+      isClickedLength: true,
+      isClickedReverse: true,
+
+    });
+  };
+
+  render() {
+    const {
+      isReversed, sortBy, isShowm, isClickedAlph,
+      isClickedLength, isClickedReverse,
+    } = this.state;
+    const visibleGoods = [...goodsFromServer];
+
+    visibleGoods.sort((a, b) => {
+      switch (sortBy) {
+        case 'length':
+          return a.length - b.length;
+
+        case 'alphabet':
+          return a.localeCompare(b);
+
+        default: return 0;
+      }
+    });
+
+    if (isReversed) {
+      visibleGoods.reverse();
+    }
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={classNames('button is-info',
+              { 'is-light': isClickedAlph })}
+            onClick={this.sortByAlphabet}
+
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={classNames('button is-success',
+              { 'is-light': isClickedLength })}
+            onClick={this.sortByLength}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={classNames('button is-warning',
+              {
+                'is-light': isClickedReverse,
+              })}
+            onClick={this.reverse}
+          >
+            Reverse
+          </button>
+
+          { isShowm
+          && (
+            <button
```

Comment:
This button shouldn't be shown if you clicked reverse 2 times (and no one sort is selected) - that causes the test to fail

Reply:
> This button shouldn't be shown if you clicked reverse 2 times (and no one sort is selected) - that causes the test to fail

there's the working version in the readme document, how it should work. 
Here is [the working version](https://mate-academy.github.io/react_list-of-goods/)

so I made it exactly as in this document.  Button reverse works absolutely the same way.
[DEMO LINK](https://vladyslava-buzova.github.io/react_list-of-goods/)
the rest of the buttons work the same way.

So I'm a bit confused about what instruction I should follow and how buttons should work

Reply:
Try to press reset 2 times on your version and on the working version
There is a difference